### DISCUSSION TITLE
Killing hung runner should output partial results to aid debugging

### DISCUSSION THREAD

**Discussion by oskarb** [nunit/nunit-console#1532] [Ideas]
I have an NUnit that has now been sitting doing nothing for 15 hours. It has successfully completed 4 test cases. I know  that it has began executing a fifth test case because the test writes a file at the beginning and I see that this has happened.

However, because the test case's output is not available until the test is complete, I cannot tell where exactly it is hanging. It can be noted that the test case _should_ have a timeout of at most 10 minutes but because I don't know which test case it's stuck on, I cannot verify that the timeout really is set.

If I kill the NUnit console process, it terminates immediately without outputting any of the output from the currently running test. It also doesn't run the cleanup code in various teardown methods (which is necessary to reset various out-of-process data).

There should be a way to tell the nunit console runner process to abort all running test cases, perform teardown, and write the output of those test cases as expected, before it dies. As it is now, because I don't know which test case or how far it got, it's very difficult to understand what's happening.

Tests are not running in parallel. Framework 3.12, console runner 3.9 (just installed 3.10 so might launch a new run soon.).

**Comment by varnk** [comment]
Surprised this request has been here for 3 years.   I am running into a similar problem.  I would like to either a) Be able to see in realtime which tests have run and passed/failed during the test run.  Or b) trap Ctrl-C so that the test runner can abort the run and report the progress thus far in the console.

**Comment by CharliePoole** [comment]
Me too. :-)

Note that option (a) is already available using the various values of the `--labels` option - see the docs. However, this doesn't help if you are trying to debug an existing run, for which you didn't use that option.

This is related to #335 and #413, which propose alternative ways to resolve the same problem. Like them, it will require major changes (probably breaking) and so is a candidate for the 4.0 release.
